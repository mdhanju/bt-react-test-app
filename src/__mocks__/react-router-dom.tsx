/* eslint-disable */

import React, { ReactNode } from 'react';

type Props = {
    children: ReactNode;
};

const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: Props): ReactNode => <div>{children}</div>;
module.exports = rrd;
