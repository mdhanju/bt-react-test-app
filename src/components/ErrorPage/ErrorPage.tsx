import React from 'react'

import ContentHolder from '../ContentHolder'
import { FormattedMessage } from 'react-intl'

const ErrorPage = (): React.ReactElement => {
    return (
        <ContentHolder>
            <FormattedMessage
                id="err.page.system.error"
                description="system Error"
            />
        </ContentHolder>
    )
}

export default ErrorPage
