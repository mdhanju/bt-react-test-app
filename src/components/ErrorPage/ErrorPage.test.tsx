/* eslint-disable react/jsx-no-undef */

import React from 'react'
import ErrorPage from '../ErrorPage'
import { MemoryRouter } from 'react-router-dom'

describe('<ErrorPage />', () => {
    it('should render initial layout', () => {
        const wrapper = shallow(<ErrorPage />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('render the error message', () => {
        const wrapper = render(
            <MemoryRouter>
                <Intl locale="en">
                    <ErrorPage />
                </Intl>
            </MemoryRouter>
        )
        expect(wrapper.text()).toBe(
            'Home - Sample FormAn error occured while loading configuration.'
        )
    })
})
