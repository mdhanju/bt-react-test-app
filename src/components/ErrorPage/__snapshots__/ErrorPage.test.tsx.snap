// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> should render initial layout 1`] = `
Array [
  <ContentHolder>
    <FormattedMessage
      description="system Error"
      id="err.page.system.error"
      values={Object {}}
    />
  </ContentHolder>,
]
`;
